common/djangoapps/student/tests/test_views.py

```diff
@@ -119,6 +120,29 @@ def test_cant_unenroll_status(self):
 
             self.assertEqual(response.status_code, 200)
 
+    def test_course_run_refund_status_successful(self):
+        """ Assert that the course refund status is 200 for successful refund call."""
+
+        with patch('student.models.CourseEnrollment.refundable', return_value=True):
+            response = self.client.get(reverse('course_run_refund_status', kwargs={'course_id': self.course.id}))
+
+        self.assertEqual(response.status_code, 200)
+
+        with patch('student.models.CourseEnrollment.refundable', return_value=False):
+            response = self.client.get(reverse('course_run_refund_status', kwargs={'course_id': self.course.id}))
+
+            self.assertEqual(response.status_code, 200)
+
+    def test_course_run_refund_status_invalid_course(self):
+        """ Assert that the course refund status is 500 for any Invalid Key ."""
+
+        with patch('opaque_keys.edx.keys.CourseKey.from_string') as mock_method:
+            mock_method.side_effect = InvalidKeyError('CourseKey', 'InvalidKeyError has occured while getting \
+                                                        refund status for a course.')
+            response = self.client.get(reverse('course_run_refund_status', kwargs={'course_id': self.course.id}))
+
+            self.assertEqual(response.status_code, 500)
```

Comment:
I think because its a server call for data and If that fails it should be raised as Internal server error.